src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,435 @@
+     */
+    private function removeExpiredLocks()
+    {
+        $now = microtime(true);
+
+        $filter = array(
+            'expires_at' => array(
+                '$lt' => $this->createDateTime($now),
+            ),
+        );
+
+        $this->getCollection()->deleteMany($filter);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException(sprintf(
+            'The store "%s" does not support blocking locks.',
+            __CLASS__
+        ));
+    }
+
+    /**
+     * @return bool
+     */
+    private function isDuplicateKeyException(
+        WriteException $e
+    ): bool {
+        $writeErrors = $e->getWriteResult()->getWriteErrors();
+        if (1 === \count($writeErrors)) {
+            $code = $writeErrors[0]->getCode();
+        } else {
+            $code = $e->getCode();
+        }
+        // Mongo error E11000 - DuplicateKey
+        return 11000 === $code;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockStorageException
+     * @throws LockExpiredException
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        $now = microtime(true);
+
+        $filter = array(
+            '_id' => (string) $key,
+            'token' => $this->getUniqueToken($key),
+            'expires_at' => array(
+                '$gte' => $this->createDateTime($now),
+            ),
+        );
+
+        $update = array(
+            '$set' => array(
+                '_id' => (string) $key,
+                'expires_at' => $this->createDateTime($now + $ttl),
+            ),
+        );
+
+        $options = array(
+            'upsert' => true,
+        );
+
+        $key->reduceLifetime($ttl);
+
+        try {
+            $this->getCollection()->updateOne($filter, $update, $options);
+        } catch (WriteException $e) {
+            if ($this->isDuplicateKeyException($e)) {
+                throw new LockConflictedException('Failed to put off the expiration of the lock', 0, $e);
+            }
+            throw new LockStorageException($e->getMessage(), 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf(
+                'Failed to put off the expiration of the "%s" lock within the specified time.',
+                $key
+            ));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(Key $key)
+    {
+        $filter = array(
+            '_id' => (string) $key,
+            'token' => $this->getUniqueToken($key),
+        );
+
+        $options = array();
+
+        $this->getCollection()->deleteOne($filter, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(Key $key): bool
+    {
+        $filter = array(
+            '_id' => (string) $key,
+            'token' => $this->getUniqueToken($key),
+            'expires_at' => array(
+                '$gte' => $this->createDateTime(),
```

Comment:
You have to define whether `equals` means locked or available. This statement conflict with `$lte` in `save`

Reply:
fixed